Add CI environment variables to user-agent

isaacs opened the pull request:
The `npm-in-ci` header has been sent for some time now by
make-fetch-happen, but User-Agent is more reliably logged and respected
by proxies and CDNs, so that's a better place to put this.

isaacs committed:
Add CI environment variables to user-agent

The `npm-in-ci` header has been sent for some time now by
make-fetch-happen, but User-Agent is more reliably logged and respected
by proxies and CDNs, so that's a better place to put this.

joeledwards commented:
This should be quite effective. Easy to parse out, and it is nice to have the additional detail regarding which CI platform is in use.

claudiahdz committed:
chore(ci): add dirPacker to options

PR-URL: https://github.com/npm/cli/pull/252
Credit: @claudiahdz
Close: #252
Reviewed-by: @mikemimik

Commit:
npm-lifecycle@3.1.4

- fix: filter functions and undefined out of makeEnv (@isaacs)

Commit:
libcipm@4.0.4

- fix: pack git directories properly (@claudiahdz)
- respect no-optional argument (@cruzdanilo)

Commit:
tar@4.4.12

Commit:
docs: updated CHANGELOG for 6.12.0

Commit:
update AUTHORS

Commit:
6.12.0-next.0

benblank committed:
Fix figgyPudding error in `npm token`

It seems that a couple of lines were missed when token.js was [changed
to use figgyPudding][1] which [causes an error][2] when trying to run
that command in certain circumstances.  This patch fixes that error.

[1]: https://github.com/npm/cli/commit/4cf850d2be118abce72b3cf533f13512e9844ad1#diff-398ed3b014436a5204583323ea29320b
[2]: https://npm.community/t/npm-token-err-figgypudding-options-cannot-be-modified-use-concat-instead/10288

PR-URL: https://github.com/npm/cli/pull/259
Credit: @benblank
Close: #259
Reviewed-by: @isaacs

isaacs committed:
node-gyp@5.0.4

PR-URL: https://github.com/npm/cli/pull/260
Credit: @isaacs
Close: #260
Reviewed-by: @isaacs
